presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -0,0 +1,365 @@
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            try (ResultSet resultSet = getTables(connection, schema, Optional.empty())) {
+                ImmutableList.Builder<SchemaTableName> list = ImmutableList.builder();
+                while (resultSet.next()) {
+                    list.add(new SchemaTableName(
+                            resultSet.getString("TABLE_SCHEM"),
+                            resultSet.getString("TABLE_NAME")));
+                }
+                return list.build();
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    protected Collection<String> listSchemas(Connection connection)
+    {
+        return ImmutableList.of(DRUID_SCHEMA);
+    }
+
+    /**
+     * Overriden to filter out tables that don't match @param schemaTableName
+     */
+    @Override
+    public Optional<JdbcTableHandle> getTableHandle(JdbcIdentity identity, SchemaTableName schemaTableName)
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            String jdbcSchemaName = schemaTableName.getSchemaName();
+            String jdbcTableName = schemaTableName.getTableName();
+            try (ResultSet resultSet = getTables(connection, Optional.of(jdbcSchemaName), Optional.of(jdbcTableName))) {
+                List<JdbcTableHandle> tableHandles = new ArrayList<>();
+                while (resultSet.next()) {
+                    tableHandles.add(new JdbcTableHandle(
+                            schemaTableName,
+                            DRUID_CATALOG,
+                            resultSet.getString("TABLE_SCHEM"),
+                            resultSet.getString("TABLE_NAME")));
+                }
+                if (tableHandles.isEmpty()) {
+                    return Optional.empty();
+                }
+                if (tableHandles.size() > 1) {
+                    return Optional.of(
+                            getOnlyElement(
+                                    tableHandles
+                                            .stream()
+                                            .filter(
+                                                    jdbcTableHandle ->
+                                                            Objects.equals(jdbcTableHandle.getSchemaName(), schemaTableName.getSchemaName())
+                                                                    && Objects.equals(jdbcTableHandle.getTableName(), schemaTableName.getTableName()))
+                                            .collect(Collectors.toList())));
+                }
+                return Optional.of(getOnlyElement(tableHandles));
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    /**
+     * Overriden since the {@link BaseJdbcClient#getTables(Connection, Optional, Optional)}
```

Comment:
Javadoc defines method behavior, here you explain technical choice. While this could go as `@implNote`, it is better to just turn this into `/*` comment 

also, Overriden -> Overridden